hack/bulk-ingest-explorer-files.py

```diff
@@ -0,0 +1,131 @@
+    if response.ok:
+        return response.json()['job_status']
+    else:
+        raise HTTPError('Bad response, got code of: {}'.format(response.status_code))
+
+
+# Get the result of a given job using the retrieveJobResult endpoint.
+def get_job_result(job_id: str):
+    response = authed_session.get(f'{jade_base_url}/api/repository/v1/jobs/{job_id}/result')
+    if response.ok:
+        return response.json()
+    else:
+        raise HTTPError(f'Failed to get the result. Got response : {response.json()}')
+
+
+# Check whether the given job is done.
+# Return true if the status is 'succeeded' or 'failed', return false if the status is 'running'.
+def is_done(job_id: str) -> bool:
+    status = check_job_status(job_id)
+    done = status in ['succeeded', 'failed']
+    if done:
+        print(f'Job {status}.')
+    return done
+
+
+# check that the response is okay, then poll the status until it finishes
+# return the final result of the job
+def wait_for_result(response):
+    if response.ok:
+        job_id = response.json()['id']
+        print(f'Waiting for job (ID: {job_id}) to finish...')
+        # check every 10 seconds until the job is finished (wait up to 48 hours)
+        poll(lambda: is_done(job_id), step=10, timeout=172800)
+        result = get_job_result(job_id)
+        return result
+    else:
+        raise HTTPError(f'Failed with response : {response.json()}')
+
+
+# Create a new (temporary) jade dataset. If successful, return the id of the dataset.
+# The kwargs should include the defaultProfileId, name, and schema to use in dataset creation.
+def create_temp_dataset(profile_id: str):
+    print('\nCreating a temporary dataset.')
+    args = {
+        'name': 'encode_explorer_ingest_test2',
+        'defaultProfileId': profile_id,
+        'schema': {'tables': [{'name': 'test_table_name', 'columns': [{'name': 'test_column', 'datatype': 'STRING'}]}]}
+    }
+    response = authed_session.post(f'{jade_base_url}/api/repository/v1/datasets', json=args)
+    result = wait_for_result(response)
+    result_dataset_id = result['id']
+    print(f'Created dataset with ID: {result_dataset_id}')
+    return result_dataset_id
+
+
+# Delete the temporary jade dataset.
+def delete_dataset(target_dataset_id: str):
+    print('\nDeleting the temporary dataset.')
+    response = authed_session.delete(f'{jade_base_url}/api/repository/v1/datasets/{target_dataset_id}')
+    wait_for_result(response)
+    print(f'Deleted dataset with ID: {target_dataset_id}')
+
+
+# submit a bulk file ingest request. If successful, return the id of the job.
+def request_bulk_file_ingest(target_dataset_id: str, load_tag: str, profile_id: str, control_file_path: str):
+    print('\nSubmitting the bulk file ingest request')
+    args = {
+        'loadTag': load_tag,
+        'profileId': profile_id,
+        'loadControlFile': control_file_path,
+        'maxFailedFileLoads': 1000000  # the jade api will eventually have 'unlimited' as the default
```

Comment:
The api currently fails when you don't specify maxFailedFileLoads. Dan D suggested that the default value for this should eventually be fixed and be updated to be "unlimited". We want the value to at least be larger than the number of files we're ingesting (so it doesn't leave the job partially finished)... I have it set to 1,000,000 right now, even though that feels a little hacky.

Reply:
Understood, so that means we're waiting on a change from Jade before we switch this ti "unlimited" or remove it completely, right?

Reply:
I'm not sure when Jade is going to fix/update it on their end, and the script works with the hard coded value, so I would be inclined to leave it as it is. If they happen to update it sooner, that would be great, but I wouldn't want to put this on hold until they do that. 

Dan D had suggested using the value of 1,000,000 for now since that's the max number of files Jade allows to be ingested in a single job.